scopesim/effects/electronic/electrons.py

```diff
@@ -91,6 +93,113 @@ def plot(self, **kwargs):
         return fig
 
 
+class InterPixelCapacitance(Effect):
+    """Inter-pixel capacitance effect.
+
+    The effect models cross-talk due to inter-pixel capacitance with
+    a convolution kernel following [1]_.
+
+    Example
+    -------
+    The effect is usually instantiated in a yaml file.
+
+    The first example uses the three-parameter model in Eq. (9) of [1].
+    The three parameters are `alpha_edge` (corresponding to $\\alpha$) for the
+    effect of neighbouring pixels sharing an edge with the pixel under
+    consideration, `alpha_corner` (corresponding to $\\alpha^\\prime$) for pixels
+    sharing a corner, and `alpha_aniso` (corresponding to $\\alpha_{+}$) to allow
+    for different capacitive coupling along rows and columns. The simpler one-
+    and two-parameters models are recovered by setting `alpha_aniso` and/or
+    `alpha_corner` to zero.
+    - name: ipc
+      description: Apply inter-pixel capacitance
+      class: InterPixelCapacitance
+      kwargs:
+         alpha_edge: 0.02
+         alpha_corner: 0.002
+         alpha_aniso: 0
+
+    Alternatively, a convolution kernel can be provided explicitely:
+    - name: ipc
+      kwargs:
+         kernel: [
+            [0.0011, 0.0127, 0.0011],
+            [0.0163, 0.9360, 0.0164],
+            [0.0011, 0.0127, 0.0011],
+          ]
+
+    .. [1] Kannawadi et al., "The Impact of Interpixel Capacitance in CMOS Detectors on PSF
+       Shapes and Implications for WFIRST", PASP 128, 095001 (2016).
+    """
+    z_order: ClassVar[tuple[int, ...]] = (810,)
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+        self.meta.update(kwargs)
+        self.kernel = self._build_kernel(kwargs)
+
+    def _build_kernel(self, params):
+        """Build a 3x3 kernel"""
+        if "kernel" in params:
+            kernel = np.asarray(params['kernel']).astype(float)
+            kernsum = np.sum(kernel)
+            if kernsum > 1:
+                logger.warning("IPC kernel is larger than one, normalising")
+                kernel /= kernsum
+            if kernsum <= 0:
+                raise ValueError("IPC kernel has negative normalisation")
+            return kernel
+
+        a_corner = params.get("alpha_corner", 0)
+        a_aniso = params.get("alpha_aniso", 0)
+        a_edge = params.get("alpha_edge", 0)
+
+        kernel = np.array([
+            [a_corner, a_edge - a_aniso, a_corner],
+            [a_edge + a_aniso, 1 - 4 * (a_edge + a_corner), a_edge + a_aniso],
+            [a_corner, a_edge - a_aniso, a_corner]
+        ])
+        return kernel
+
+    def apply_to(self, det, **kwargs):
+        if not isinstance(det, Detector):
+            logger.debug("%s applied to %s", self.display_name,
+                         det.__class__.__name__)
+            return det
+
+        newdata = convolve(det._hdu.data, self.kernel, mode="same")
```

Comment:
Perhaps use `oaconvolve()`? It is [supposedly faster in this scenario](https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.oaconvolve.html), but I have no experience myself:

> This is generally much faster than [convolve](https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.convolve.html#scipy.signal.convolve) for large arrays (n > ~500), and generally much faster than [fftconvolve](https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.fftconvolve.html#scipy.signal.fftconvolve) when one array is much larger than the other,